src/command/backup/backup.c

```diff
@@ -0,0 +1,1911 @@
+            lstSort(*(List **)lstGet(*queueList, targetIdx), sortOrderDesc);
+
+        // Move process queues to calling context
+        lstMove(*queueList, MEM_CONTEXT_OLD());
+    }
+    MEM_CONTEXT_TEMP_END();
+
+    FUNCTION_LOG_RETURN(UINT64, result);
+}
+
+// Helper to caculate the next queue to scan based on the client index
+static int
+backupJobQueueNext(unsigned int clientIdx, int queueIdx, unsigned int queueTotal)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(UINT, clientIdx);
+        FUNCTION_TEST_PARAM(INT, queueIdx);
+        FUNCTION_TEST_PARAM(UINT, queueTotal);
+    FUNCTION_TEST_END();
+
+    // Move (forward or back) to the next queue
+    queueIdx += clientIdx % 2 ? -1 : 1;
+
+    // Deal with wrapping on either end
+    if (queueIdx < 0)
+        FUNCTION_TEST_RETURN((int)queueTotal - 1);
+    else if (queueIdx == (int)queueTotal)
+        FUNCTION_TEST_RETURN(0);
+
+    FUNCTION_TEST_RETURN(queueIdx);
+}
+
+// Callback to fetch backup jobs for the parallel executor
+typedef struct BackupJobData
+{
+    const String *const backupLabel;                                // Backup label (defines the backup path)
+    const bool backupStandby;                                       // Backup from standby
+    const String *const cipherSubPass;                              // Passphrase used to encrypt files in the backup
+    const bool compress;                                            // Is the backup compressed?
+    const unsigned int compressLevel;                               // Compress level if backup is compressed
+    const bool delta;                                               // Is this a checksum delta backup?
+    const uint64_t lsnStart;                                        // Starting lsn for the backup
+
+    List *queueList;                                                // List of processing queues
+} BackupJobData;
+
+static ProtocolParallelJob *backupJobCallback(void *data, unsigned int clientIdx)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM_P(VOID, data);
+        FUNCTION_TEST_PARAM(UINT, clientIdx);
+    FUNCTION_TEST_END();
+
+    ASSERT(data != NULL);
+
+    ProtocolParallelJob *result = NULL;
+
+    MEM_CONTEXT_TEMP_BEGIN()
+    {
+        // Get a new job if there are any left
+        BackupJobData *jobData = data;
+
+        // Determine where to begin scanning the queue (we'll stop when we get back here).  When copying from the primary during
+        // backup from standby only queue 0 will be used.
+        unsigned int queueOffset = jobData->backupStandby && clientIdx > 0 ? 1 : 0;
+        int queueIdx = jobData->backupStandby && clientIdx == 0 ?
+            0 : (int)(clientIdx % (lstSize(jobData->queueList) - queueOffset));
+        int queueEnd = queueIdx;
+
+        do
+        {
+            List *queue = *(List **)lstGet(jobData->queueList, (unsigned int)queueIdx + queueOffset);
+
```

Comment:
This was a bug in Perl.  We were getting the copyStart from the remote system but Start and Stop from the local system when online.  If the clocks are not synced then it could get weird.

All timestamps should come from the database host (which could be local) when online and directly from time() when offline.